org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -703,4 +709,44 @@ public void projectsBuildFinished(IProgressMonitor monitor) {
 	public boolean isBuildFinished() {
 		return buildFinished;
 	}
+
+	@Override
+	public void checkIndexes() {
+		Job job = new Job(("Check Indexes")) {
+
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				IJavaProject[] javaProjects = ProjectUtils.getJavaProjects();
+				if (javaProjects.length <= 0) {
+					return Status.OK_STATUS;
+				}
+				SearchEngine engine = new SearchEngine();
+				IJavaSearchScope scope = SearchEngine.createWorkspaceScope();
+				boolean found[] = { false };
+				try {
+					engine.searchAllTypeNames("java.lang".toCharArray(), SearchPattern.R_EXACT_MATCH, "String".toCharArray(), //$NON-NLS-1$
+							SearchPattern.R_PATTERN_MATCH | SearchPattern.R_CASE_SENSITIVE, IJavaSearchConstants.CLASS, scope, new TypeNameRequestor() {
+								@Override
+								public void acceptType(int modifiers, char[] packageName, char[] simpleTypeName, char[][] enclosingTypeNames, String path) {
+									found[0] = true;
+								}
+							}, IJavaSearchConstants.WAIT_UNTIL_READY_TO_SEARCH, monitor);
+				} catch (JavaModelException e) {
+					// search failed
+				} catch (OperationCanceledException e) {
+					return Status.CANCEL_STATUS;
+				}
+				if (!found[0]) {
+					try {
+						JavaLanguageServerPlugin.logInfo("rebuild indexes");
```

Comment:
"Unable to locate JDK types through index. Attempting to rebuild all indexes."

Reply:
Fixed.